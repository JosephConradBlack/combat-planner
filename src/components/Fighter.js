import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Segment, Item, Header, Image, Responsive } from 'semantic-ui-react';

import FighterIcon from './FighterIcon';
import NounIcon from './NounIcon';
import placeholderFighter from '../assets/placeholderfighter.svg';

const HeaderContainer = styled(Item.Header)`
  &&& {
    display: inline-block;
  }
`;

const HeaderContainerWithEllipsis = styled(HeaderContainer)`
  @media only screen and (max-width: 500px) {
    width: 170px;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
  }
`;

const StyledPlaceholerImage = styled(Image)`
  opacity: 0.8;
`;

const StyledResponsive = styled(Responsive)`
  &&& {
    display: inline;
  }
`;

const Fighter = class extends React.Component {
  static propTypes = {
    index: PropTypes.number,
    item: PropTypes.object,
    handleChange: PropTypes.func
  };

  constructor(props) {
    super(props);

    this.state = { health: props.item ? props.item.health : null };
  }

  handleHealthChange = (event, data) => {
    this.setState({ health: data.value });

    if (this.props.handleChange) {
      this.props.handleChange({
        ...this.props.item,
        health: data.value
      });
    }
  };

  getFighter = (index, item) => (
    <Item.Content>
      <Item.Extra>
        {item.initiative && (
          <React.Fragment>
            <NounIcon src="/icon_sword.svg" floated="left" />
            <Header floated="left">{item.initiative}</Header>
          </React.Fragment>
        )}
      </Item.Extra>
      <HeaderContainerWithEllipsis>{item.name}</HeaderContainerWithEllipsis>
      {item.health && (
        <FighterIcon
          icon="heart outline"
          color="red"
          number={this.state.health}
          position={1}
          handleChange={this.handleHealthChange}
        />
      )}
      {item.armor && (
        <FighterIcon
          icon="shield"
          color="grey"
          number={item.armor}
          position={0}
          readonly
        />
      )}
    </Item.Content>
  );

  getEmptyFighter = () => (
    <Item.Content>
      <Item.Extra>
        <StyledPlaceholerImage floated="left" src={placeholderFighter} />
      </Item.Extra>
      <HeaderContainer />
      <StyledResponsive minWidth={600}>
        <FighterIcon
          icon="heart outline"
          color="grey"
          number={null}
          position={1}
        />
        <FighterIcon icon="shield" color="grey" number={null} position={0} />
      </StyledResponsive>
    </Item.Content>
  );

  render() {
    const { className, index, item } = this.props;
    return (
      <Segment className={className}>
        <Item>
          {item ? this.getFighter(index, item) : this.getEmptyFighter()}
        </Item>
      </Segment>
    );
  }
};

export default Fighter;
