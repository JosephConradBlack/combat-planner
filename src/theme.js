export default {
  h1: {
    fontSize: '6em',
    fontWeight: 300,
    letterSpacing: '0.06em',
    subheader: {
      fontSize: '1em',
      fontWeight: 'lighter'
    }
  }
};
