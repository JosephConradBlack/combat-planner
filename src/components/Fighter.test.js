import React from 'react';
import { create } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';
import Fighter from './Fighter';

describe('Fighter', () => {
  it('renders with no props', () => {
    const component = create(
      <ThemeProvider theme={theme}>
        <Fighter />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with fighter', () => {
    const item = {
      id: 1,
      name: 'Fighter'
    };
    const component = create(
      <ThemeProvider theme={theme}>
        <Fighter item={item} />
      </ThemeProvider>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
