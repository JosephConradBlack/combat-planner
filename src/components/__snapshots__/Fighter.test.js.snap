// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fighter renders with fighter 1`] = `
<Segment
  color=""
  placeholder={false}
>
  <Item>
    <ItemContent>
      <ItemExtra />
      <Styled(Styled(ItemHeader))>
        Fighter
      </Styled(Styled(ItemHeader))>
    </ItemContent>
  </Item>
</Segment>
`;

exports[`Fighter renders with no props 1`] = `
<Segment
  color=""
  placeholder={true}
>
  <Item>
    <ItemContent>
      <ItemExtra>
        <Styled(Image)
          floated="left"
          src="placeholderfighter.svg"
        />
      </ItemExtra>
      <Styled(ItemHeader) />
      <Styled(Responsive)
        minWidth={600}
      >
        <FighterIcon
          color="grey"
          icon="heart outline"
          number={null}
          position={1}
          readonly={false}
        />
        <FighterIcon
          color="grey"
          icon="shield"
          number={null}
          position={0}
          readonly={false}
        />
      </Styled(Responsive)>
    </ItemContent>
  </Item>
</Segment>
`;
